anydex/wallet/iota/iota_wallet.py

```diff
@@ -0,0 +1,334 @@
+
+        # Get wallet balance
+        balance = await self.get_balance()
+
+        if balance['available'] < value:
+            return InsufficientFunds(f'Balance {balance["available"]} of the wallet is less than {value}.')
+
+        # generate and send a transaction
+        transaction = ProposedTransaction(
+            address=Address(address),
+            value=value
+        )
+
+        # Submit the transaction to the tangle
+        self._logger.info(f"Creating {self.network} payment with amount {value} to address {address}")
+        bundle = await self.provider.submit_transaction(transaction)
+
+        # Return bundle hash ID instead of transaction ID
+        return bundle.hash.__str__()
+
+    async def get_balance(self):
+        """
+        Fetch the balance of the wallet: of all addresses specified for the seed
+        :return: available balance, pending balance, currency, precision
+        """
+        if not self.created:
+            return {
+                'available': 0,
+                'pending': 0,
+                'currency': self.get_identifier(),
+                'precision': self.precision()
+            }
+
+        # update bundles and transactions database
+        await self.update_bundles_database()
+        await self.update_transactions_database()
+
+        # fetch available and pending balance
+        available = await self.provider.get_seed_balance()
+        pending = await self.get_pending()
+
+        return {
+            'available': available,
+            'pending': pending,
+            'currency': self.get_identifier(),
+            'precision': self.precision()
+        }
+
+    async def get_pending(self):
+        """
+        Get the pending balance of the given seed
+        :return: the pending balance
+        """
+        if not self.created:
+            return 0
+
+        # fetch pending transactions from the database
+        pending_transactions = self.database.query(DatabaseTransaction) \
+            .filter(DatabaseTransaction.seed.__eq__(self.seed.__str__())) \
+            .filter(DatabaseTransaction.is_confirmed.is_(False)) \
+            .all()
+
+        # add up values of all the pending transactions
+        pending_balance = sum([tx.value for tx in pending_transactions])
+
+        return pending_balance
+
+    async def get_transactions(self):
+        """
+        Fetch the transactions related to the seed through the API and store them
+        :return:
+        """
+        if not self.created:
+            return []
+
+        # update bundles and transactions database
+        await self.update_bundles_database()
+        await self.update_transactions_database()
+
+        # get all transactions of this seed
+        db_seed_transactions = self.database.query(DatabaseTransaction) \
+            .filter(DatabaseTransaction.seed.__eq__(self.seed.__str__())) \
+            .all()
+        db_addresses = self.database.query(DatabaseAddress).all()
```

Comment:
Wouldn't this also include testnet addresses

Reply:
Fixed it in the refactor branch to only fetch addresses related to the seed currently being used 👍